plugins/module_utils/controller.py

```diff
@@ -383,3 +383,31 @@ def delete_if_needed(
             raise EDAError(msg)
         msg = f"Unable to delete {item_name}: {response.status}"
         raise EDAError(msg)
+
+    def copy_if_needed(
+        self,
+        name: str,
+        copy_from: str,
+        endpoint: str,
+        item_type: str = "unknown",
+    ) -> dict[str, bool]:
+        response = None
+        if not copy_from:
+            msg = f"Unable to copy {item_type}, missing copy_from parameter"
+            raise EDAError(msg)
```

Comment:
I think this can be removed because copy_if_needed() will be called only when copy_from is defined.